### DISCUSSION TITLE
Win32 - Working Project Opener / Editor

### DISCUSSION THREAD

**Discussion by Ambushfall** [johnlindquist/kit#1067] [Share]
[Open dev-project in Script Kit](https://scriptkit.com/api/new?name=dev-project&url=https://gist.githubusercontent.com/Ambushfall/6bf540b63a5b4fc942d65deaec04f149/raw/5f0d048c608dc35558c8b661cb6c9c455c69efcb/dev-project.js")

```js
// Preview: docs
// Menu: Open Project
// Description: Opens a project in vscode
// Shortcut: cmd shift .

import "@johnlindquist/kit";

const envPath = await env('PROJECT_DIR');

const projectDir = home(envPath);

const projects = await readdir(projectDir);

const filteredProjects = projects.filter((v) => v.includes('.') ? false : v)

const project = await arg('Enter a project name:', filteredProjects);

const projectPath = path.resolve(projectDir, project);

edit(projectPath)
```

# Todos
- [ ] Make it so editor does not create an untitled file

**Comment by johnlindquist** [comment]
@Ambushfall 

Try this instead of `edit`. "edit" expects a file:

```js
await $`code ${projectPath}`
```

**Reply by Ambushfall** [reply]
@johnlindquist 

Sorry, this unfortunately does not work, due to the way $ handles injected string paths on windows.

What's the logical desired outcome? Open the project folder with vscode/default editor rather than opening a file directly, in case the path is a file, open the file instead.

Sure it's possible to make a check isDir() within the edit wrapper.

See this discussion for more info:
[Discussion](https://github.com/johnlindquist/kit/discussions/1058)
